### PR TITLE
Add additional boost guards

### PR DESCRIPTION
**Pull request by varunagrawal**
As the title says.

### CODE REVIEW

**Review by arutkowski — APPROVED** [review]
Thanks @varunagrawal, I should have caught these when working on #378 (clearly, I skipped running the tests). Everything builds and all the unit tests pass on my end with boost disabled, so I'll call it good!

**Comment by arutkowski** [comment]
@varunagrawal, it doesn't feel right to me to actually hit the merge button since I'm not in your lab. I'll leave that to you.

**Comment by varunagrawal** [comment]
Thanks for the review @arutkowski

**Pull request merged**
